### PR TITLE
Version 0.4

### PR DESCRIPTION
**Pull request by SebastianSemper**
Let's get this thing going.

### CODE REVIEW

File: eadf/importers.py [hunk 1]
```diff
@@ -186,3 +189,53 @@ def fromAngleListData(
     )
 
     return EADF(arrInter, arrAzi, arrCoEle, arrFreqData, arrPos)
+
+def loadData(
```
Comment (SebastianSemper) [inline eadf/importers.py]: The function name is completely void of information. It should be changed to something more meaningful.

File: eadf/importers.py [hunk 2]
```diff
@@ -186,3 +189,53 @@ def fromAngleListData(
...
+    path: str,
+    numPor = 0,
```
Comment (SebastianSemper) [inline eadf/importers.py]: why is it the number of ports, and not an array of port indices? why is this feature only there for the antenna ports and not also for frequency and polarisation? also the specification of the default value is not pep8-conform.

File: eadf/importers.py [hunk 3]
```diff
@@ -186,3 +189,53 @@ def fromAngleListData(
...
+) -> EADF:
+    """
+    This format defines angular data over uniformly sampled
+    ele, azi, elem, pol, and freq. Using the importer it is possible to choose
+    the number of ports to derive the EADF by supplying the numPor value.
```
Comment (SebastianSemper) [inline eadf/importers.py]: why must it be uniformly sampled along frequency and polarisation? what does that even mean? also, there is no description about the internal structure of the expected matfile. how would someone be able to produce a .mat-file that can be processed by this importer? however, this is crucially necessary.

File: eadf/importers.py [hunk 4]
```diff
@@ -186,3 +189,53 @@ def fromAngleListData(
...
+    arrCoEle = (
+        np.radians(data["pattern"][0][0][0][0][0][1].flatten()) + np.pi / 2
+    )
+    arrAzi = np.radians(data["pattern"][0][0][0][0][1][1].flatten())
+    arrFreq = data["pattern"][0][0][0][0][4][1]
+    arrPattern = data["pattern"][0][0][1]
+    numPorData = (data["pattern"][0][0][0][0][2][1]).size
+
+    if (numPor > numPorData) or (numPor < 1):
+        numPor = 9
```
Comment (SebastianSemper) [inline eadf/importers.py]: what? this looks wrong. either throw an error if the function parameters are invalid, or at least issue a warning, that we did something different than the user specified.

**Review by SebastianSemper — COMMENTED** [review]
the current state of loadData is not acceptable. not only does it fail the pycodestyle checks, but it also is not well documented and does not possess enough functionality as one might wish.

File: eadf/importers.py (hunk 3 above)
Comment (pawarsp) [reply eadf/importers.py]: by this did you mean that the details about the measurement method itself are lacking?

File: eadf/importers.py (hunk 4 above)
Comment (pawarsp) [reply eadf/importers.py]: that is certainly incorrect.

File: eadf/importers.py (hunk 2 above)
Comment (pawarsp) [reply eadf/importers.py]: Agree that the user should be able to choose the specific antenna ports, polarization and frequency points.

**Comment by SebastianSemper** [comment]
No, I mean that as a user one does not know what should be in the mat-file
such that the importer works. What objects have to be there in Matlab? How
should they be named? Their dimensions? Their datatype? In general: what is
the exakt file format description for the files your importer imports?

Sankalp Pawar <notifications@github.com> schrieb am Do., 6. Feb. 2020,
16:52:

**Review by pawarsp — APPROVED** [review]

**Pull request merged**